src/containers/ProfilePage/ProfilePage.test.js

```diff
@@ -161,3 +185,153 @@ describe('ProfilePage', () => {
     expect(screen.getAllByTitle('3/5')).toHaveLength(2);
   });
 });
+
+describe('Duck', () => {
+  const config = {
+    ...getHostedConfiguration(),
+    accessControl: { marketplace: { private: true } },
+  };
+
+  // Shared parameters for viewing rights loadData tests
+  const fakeResponse = resource => ({ data: { data: resource, include: [] } });
+  const sdkFn = response => jest.fn(() => Promise.resolve(response));
+  const forbiddenError = new Error({ status: 403, message: 'forbidden' });
+  const errorSdkFn = error => jest.fn(() => Promise.reject(error));
+
+  it("loadData() for full viewing rights user loads someone else's profile", () => {
+    const initialState = getInitialState();
+
+    const { currentUser } = initialState.user;
+    const { reviews, userListingRefs } = initialState.ProfilePage;
+    const { l1: listing } = initialState.marketplaceData.entities.listing;
+    const { userId: user } = initialState.marketplaceData.entities.user;
+
+    const getState = () => ({
+      ...initialState,
+      auth: { isAuthenticated: true },
+    });
+
+    const sdk = {
+      currentUser: { show: sdkFn(fakeResponse(currentUser)) },
+      users: { show: sdkFn(fakeResponse(user)) },
+      reviews: { query: sdkFn(fakeResponse(reviews)) },
+      listings: { query: sdkFn(fakeResponse([listing])) },
+      authInfo: sdkFn({}),
+    };
+
+    const dispatch = createFakeDispatch(getState, sdk);
+
+    // This is now sanitizeConfig is parsed in the showUser thunk
+    const userFields = config?.user?.userFields;
+    const sanitizeConfig = { userFields };
+
+    // Tests the actions that get dispatched to the Redux store when ProfilePage.duck.js
+    // loadData() function is called. If you make customizations to the loadData() logic,
+    // update the dispatched actions list in this test accordingly!
+    return loadData({ id: userId }, null, config)(dispatch, getState, sdk).then(data => {
+      expect(dispatchedActions(dispatch)).toEqual([
+        setInitialState(),
+        currentUserShowRequest(),
+        showUserRequest(user.id, config),
+        queryListingsRequest(user.id),
+        currentUserShowSuccess(currentUser),
+        addMarketplaceEntities(fakeResponse(user), sanitizeConfig),
+        showUserSuccess(),
+        addMarketplaceEntities(fakeResponse([listing])),
+        queryListingsSuccess(userListingRefs),
+        queryReviewsSuccess(reviews),
```

Comment:
There are some surprises. (Old code is missing queryReviewsRequest. It might make sense to fix it at some point but perhaps not inside this PR.)